src/CsvHelper/CsvParser.cs

```diff
@@ -160,6 +160,16 @@ public string[] Record
 		/// <param name="configuration">The configuration.</param>
 		public CsvParser(TextReader reader, CsvConfiguration configuration)

```

Comment:
Hi!
Defensive programming is great. What about marking parameters with `System.Diagnostics.CodeAnalysis.NotNullAttribute`?
Something like this:
```suggestion
		public CsvParser([NotNull] TextReader reader, [NotNull] CsvConfiguration configuration)
```

Reply:
Thanks @zetroot. I'm generally in favor of Nullable static analysis, but CsvHelper targets .NET Framework 4.5/4.7, as well as .NET Standard 2.0, which would make it tricky (if at all possible) to enable that.

Perhaps something that could be added as a follow-up PR to enable nullable checks (with any necessary polyfills) would be a good idea if @JoshClose would be open to it.

Reply:
@augustoproiete, you are absoluetly right! Lagecy platforms will spoil us a lot of blood. 
It seems it could be fixed with `#pragma`, but it will be a big job. This change must be supported  by ci/cd and so on